src/main/java/com/gluonhq/substrate/target/AndroidTargetConfiguration.java

```diff
@@ -150,23 +179,28 @@ public boolean link() throws IOException, InterruptedException {
         Files.createDirectories(dalvikBinPath);
         Files.createDirectories(dalvikLibPath);
         Files.createDirectories(dalvikLibArm64Path);
-        Path androidManifestPath = dalvikPath.resolve("AndroidManifest.xml");
+        Path androidManifestDalvikPath = dalvikPath.resolve(Constants.MANIFEST_FILE);
         Path dalvikActivityPackage = dalvikClassPath.resolve("com/gluonhq/helloandroid");
         Path dalvikKeyCodePackage = dalvikClassPath.resolve("javafx/scene/input");
         FileOps.copyResource("/native/android/dalvik/MainActivity.class", dalvikActivityPackage.resolve("MainActivity.class"));
         FileOps.copyResource("/native/android/dalvik/MainActivity$1.class", dalvikActivityPackage.resolve("MainActivity$1.class"));
         FileOps.copyResource("/native/android/dalvik/MainActivity$InternalSurfaceView.class", dalvikActivityPackage.resolve("MainActivity$InternalSurfaceView.class"));
         FileOps.copyResource("/native/android/dalvik/KeyCode.class", dalvikKeyCodePackage.resolve("KeyCode.class"));
         FileOps.copyResource("/native/android/dalvik/KeyCode$KeyCodeClass.class", dalvikKeyCodePackage.resolve("KeyCode$KeyCodeClass.class"));
-        FileOps.copyResource("/native/android/AndroidManifest.xml", dalvikPath.resolve("AndroidManifest.xml"));
-        FileOps.replaceInFile(dalvikPath.resolve("AndroidManifest.xml"), "package='com.gluonhq.helloandroid'", "package='" + projectConfiguration.getAppId() + "'");
-        FileOps.replaceInFile(dalvikPath.resolve("AndroidManifest.xml"), "A HelloGraal", projectConfiguration.getAppName());
 
-        // resources
-       for (String iconFolder : iconFolders) {
+        Path androidManifest = androidPath.resolve(Constants.MANIFEST_FILE);
+        if (!Files.exists(androidManifest)) {
+            throw new IOException("File " + androidManifest.toString() + " not found");
+        }
+        FileOps.copyFile(androidManifest, androidManifestDalvikPath);
+        for (String iconFolder : iconFolders) {
+            Path iconPath = androidPath.resolve("assets").resolve("res").resolve(iconFolder).resolve("ic_launcher.png");
+            if (!Files.exists(iconPath)) {
+                throw new IOException("File " + iconPath.toString() + " not found");
```

Comment:
Is it really critical for an icon image to exist?

Reply:
Yes, `aapt package` will fail if these icons are not there.

Reply:
OK.